docs/source/conf.py

```diff
@@ -0,0 +1,159 @@
+# else:
+#     gh_source_version = version
+
+# # define function for resolving source link
+# def linkcode_resolve(domain, info):
+#     if domain != 'py':
+#         return None
+#     if not info['module']:
+#         return None
+#     filename = info['module'].replace('.', '/')
+#     return "https://github.com/labscript-suite/{}/blob/{}/{}.py".format(project, gh_source_version, filename)
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = []
+
+# The suffix(es) of source filenames.
+source_suffix = ['.rst', '.md']
+
+# The master toctree document.
+master_doc = 'index'
+
+# intersphinx allows us to link directly to other repos sphinxdocs.
+# https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3/', None),
+    'numpy': ('https://numpy.org/doc/stable/', None),
+    'scipy': ('https://docs.scipy.org/doc/scipy/reference/', None),
+    'pandas': ('https://pandas.pydata.org/pandas-docs/stable/', None),
+    'qtutils': ('https://qtutils.readthedocs.io/en/stable/', None),
+    'pyqtgraph': ('https://pyqtgraph.readthedocs.io/en/latest/', None), # change to stable once v0.11 is published
+    'matplotlib': ('https://matplotlib.org/', None),
+    'h5py': ('http://docs.h5py.org/en/stable/', None),
+    'pydaqmx': ('https://pythonhosted.org/PyDAQmx/', None),
+    'qt': ('', 'pyqt5-modified-objects.inv') # from https://github.com/MSLNZ/msl-qt/blob/master/docs/create_pyqt_objects.py under MIT License
+    # TODO
+    # desktop-app
+    # spinapi/pynivision/etc
+}
+
+# list of all labscript suite components that have docs
+labscript_suite_programs = [
+    'labscript',
+    'runmanager',
+    'runviewer',
+    'blacs',
+    'lyse',
+    'labscript-utils',
+    'labscript-devices',
+]
+# remove this current repo from the list
+if project in labscript_suite_programs:
+    labscript_suite_programs.remove(project)
+
+# whether to use stable or latest version
+labscript_suite_doc_version = 'stable' # 'stable' or 'latest'
+
+# add intersphinx references for each component
+for ls_prog in labscript_suite_programs:
+    intersphinx_mapping[ls_prog] = ('https://docs.labscript_suite.org/projects/{}/en/{}/'.format(ls_prog, labscript_suite_doc_version), None)
+
+# add intersphinx reference for the metapackage
+intersphinx_mapping['labscript-suite'] = ('https://docs.labscript_suite.org/en/{}/'.format(labscript_suite_doc_version), None)
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = "sphinx_rtd_theme"
+html_logo = "img/{}.svg".format(project)
```

Comment:
These can be absolute links to the art folder of labscript-suite/labscript-suite#38. Is the idea here that we'll duplicate these in each repository?

Reply:
Oh, right. They could be links to the files hosted on GitHub directly! I had been trying to work out how to link to them assuming they could somehow be hosted by RTD, but couldn't figure out the solution!

Reply:
Yep, this should do it:
```python
html_logo = "https://raw.githubusercontent.com/labscript-suite/labscript-suite/master/art/labscript-suite-rectangular-transparent_276x140.svg"
html_favicon = "https://raw.githubusercontent.com/labscript-suite/labscript-suite/master/art/labscript.ico"
```

Reply:
So that doesn't work. Apparently the path has to be [relative to the config directory](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_logo) :(

Reply:
:(

How about getting them at build, using a [sphinx core event](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx-core-events) handler? This worked for me locally:

```python
from pathlib import Path
import requests

collect_assets = {
    "https://raw.githubusercontent.com/labscript-suite/labscript-suite/master/art/labscript-suite-rectangular-transparent_276x140.svg": html_logo,
    "https://raw.githubusercontent.com/labscript-suite/labscript-suite/master/art/labscript.ico": html_favicon,
}


def config_inited_handler(app, config):
    for url, file_name in collect_assets.items():
        r = requests.get(url, allow_redirects=True)
        out_path = Path(__file__).parents[0] / file_name
        out_path.parents[0].mkdir(parents=True, exist_ok=True)
        with open(out_path, "wb") as out_file:
            out_file.write(r.content)


def setup(app):
    ...
    app.connect("config-inited", config_inited_handler)
```

Reply:
Alternatively:
```python
from pathlib import Path
import requests

html_logo = "art/labscript-suite-rectangular-transparent_276x140.svg"
html_favicon = "art/labscript.ico"

raw_base = "https://raw.githubusercontent.com"
owner = "labscript-suite"
repository = "labscript-suite"
branch = "master"

collect_assets = [html_logo, html_favicon]


def config_inited_handler(app, config):
    for asset in collect_assets:
        url = "/".join([raw_base, owner, repository, branch, asset])
        r = requests.get(url, allow_redirects=True)
        out_path = Path(__file__).parents[0] / asset
        out_path.parents[0].mkdir(parents=True, exist_ok=True)
        with open(out_path, 'wb') as out_file:
            out_file.write(r.content)


def setup(app):
    ...
    app.connect("config-inited", config_inited_handler)
```

Reply:
I've thought about this and don't really like dynamically pulling in resources at build time. It leads to inconsistent build results. So for the moment I'm leaving as-is. We can revisit if you think it's worth it, but I suggest it's a separate PR as it doesn't need to hold up the rollout of the initial docs.